services/users/src/services/user.service.js

```diff
@@ -0,0 +1,51 @@
+import bcrypt from 'bcryptjs';
+import { v4 as uuid } from 'uuid';
+import { User } from '../db/models/user.model.js';
+import { errorMessages, SALT } from '../config.js';
+
+class UserService {
+  async getUsers() {
+    return await User.scope('withoutPassword').findAll({ order: ['id'] });
+  }
+
+  async getUserById(id) {
+    const user = await User.findOne({ where: { id } });
+
+    if (!user) throw new Error(errorMessages.USER_NOT_EXISTS_ID);
+
+    return user;
+  }
+
+  async getUserByEmail(email) {
+    return await User.findOne({ where: { email } });
+  }
+
+  async createUser(userData) {
+    userData.id = uuid();
```

Comment:
It's a bad practice to modify function arguments, I guess you should firstly copy userData 
```js
const user = {...userData}
``` 
and then add new fields

Reply:
Fixed